File: LeetCode/LeetCode 1221. 分割平衡字符串.cpp
class Solution {
public:
    int balancedStringSplit(string s) {
        int ans = 0, sum_l = 0, sum_r = 0;
        for(auto &x : s) {
            if(x == 'L') sum_l++;
            else sum_r++;
            if(sum_l == sum_r) ans++;
        }
        return ans;
    }
};

File: LeetCode/LeetCode 1221. 分割平衡字符串.py
class Solution:
    def balancedStringSplit(self, s: str) -> int:
        ans = sum_l = sum_r = 0
        for i in range(len(s)):
            if s[i] == 'L':
                sum_l += 1
            else:
                sum_r += 1
            if sum_l == sum_r:
                ans += 1

        return ans